File: lab6.c
#include<stdio.h>
#include<stdlib.h>
void E();
int i = 0;
char str[20],tp;
void advance()
{
i++;
tp=str[i];
}
void F()
{
if(tp=='a')
{
advance();
}
else
{
if(tp=='(')
{
advance();
E();
if(tp==')')
{
advance();
}
else
{
printf("String is not accepted\n");
exit(1);
}
}
else {
printf("\nString is not accepted");
exit(1);
}
}
}
void TP()
{
if(tp=='*')
{
advance();

F();
TP();
}
}
void T()
{
F();
TP();
}
void EP()
{
if(tp=='+')
{
advance();
T();
EP();
}
}

void E()
{
T();
EP();
}
int main()
{
printf("Enter the string: ");
scanf("%s", &str[i]);
tp = str[i];
E();
if(tp=='\0')
{
printf("\nString is accepted");
}
else
{
printf("\nString is not accepted");
}
exit(0);
}

File: SSC/lab6.py
i = 0
tp = ''
s = ''

def advance():
    global i, tp, s
    i += 1
    if (len(s) <= i):
        return
    else:
        tp = s[i]

def F():
    global tp, i, s
    if (tp == 'a'):
        advance()
    else:
        if (tp == '('):
            advance()
            E()
            if (tp == ')'):
                advance()
            else:
                print("Not accepted\n")
                exit(1)
        else:
            print("\nAccepted")
            exit(1)

def TP():
    global tp, i, s
    if (tp == '*'):
        advance()
        F()
        TP()

def T():
    global tp, i, s
    F()
    TP()

def EP():
    global tp, i, s
    if (tp == '+'):
        advance()
        T()
        EP()

def E():
    global tp, i, s
    T()
    EP()

def main():
    global tp, i, s
    s = input("String? ")
    tp = s[i]
    E()

    if (tp == s[-1]):
        print("\nAccepted")
    else:
        print("\nNOT accepted")
    
    exit(0)

main()